src/custom/state/orders/utils.ts

```diff
@@ -219,8 +220,12 @@ export function getEstimatedExecutionPrice(
 
   // Build CurrencyAmount and Price instances
   const feeAmount = CurrencyAmount.fromRawAmount(order.inputToken, fee)
-  const inputAmount = CurrencyAmount.fromRawAmount(order.inputToken, order.sellAmount.toString())
-  const outputAmount = CurrencyAmount.fromRawAmount(order.outputToken, order.buyAmount.toString())
+  const inputAmount = CurrencyAmount.fromRawAmount(order.inputToken, getRemainingAmount('sell', order))
+  const outputAmount = CurrencyAmount.fromRawAmount(order.outputToken, getRemainingAmount('buy', order))
```

Comment:
This is strange. Isn't the function meant to be used so you pass `sell` if its a sell order (`buy` otherwise), but here you pass both

Reply:
True.
I should use the amounts received from the quote for the opposite amount. 
Although that won't be the new limit price.
For this then I think I should get the remainder opposite amount discounting the surplus.